sql/core/src/test/scala/org/apache/spark/sql/QueryTest.scala

```diff
@@ -274,6 +274,13 @@ abstract class QueryTest extends PlanTest {
     val normalized1 = logicalPlan.transformAllExpressions {
       case udf: ScalaUDF => udf.copy(function = null)
       case gen: UserDefinedGenerator => gen.copy(function = null)
+      // SPARK-17356: In usage of mllib, Metadata may store a huge vector of data, transforming
+      // it to JSON may trigger OutOfMemoryError.
+      case a @ Alias(child, name) if a.explicitMetadata.isDefined =>
+        Alias(child, name)(a.exprId, a.qualifier, Some(Metadata.empty), a.isGenerated)
+      case a: AttributeReference if a.metadata != Metadata.empty =>
+        AttributeReference(a.name, a.dataType, a.nullable, Metadata.empty)(a.exprId, a.qualifier,
+          a.isGenerated)
```

Comment:
@yhuai 

It is covered by existing QueryTest. In QueryTest, it will call toJSON and then call fromJSON to convert JSON back to a plan, and do comparison like this:
```
    if (normalized1 != normalized2) {
      fail(
        s"""
           |== FAIL: the logical plan parsed from json does not match the original one ===
           |${sideBySide(logicalPlan.treeString, normalized2.treeString).mkString("\n")}
          """.stripMargin)
    }
```

If the Meta data is not empty, the comparison will fail.